src/stories/README/Table.md

```diff
@@ -19,6 +19,26 @@ The `columns` property should be an array of objects with the following properti
 | render | <code>(value: any, row: T) => string &#124; number &#124; number &#124; JSX.Element &#124; null</code> | - | Use a custom render function to display the value in each column cell. This function will be called with the value of the `field` provided and the row data (`T`) |
 | sortable | <code>boolean &#124; (a: T, b: T) => number</code> | - | If true, the column will be sortable using an alphanumeric sort, alternatively a function can be provided allowing finer grained control over sorting |
 
+## Notes
+
+For performance reasons table rows are only re-rendered if the row properties
+have changed. For this reason, rows **will not** re-render if data is mutated:
+you should clone the item instead, for example:
+
+```js
+update (index) {
+  const newData = this.state.data
+  const element = newData[index]
+  newData[index] = Object.assign({}, element, { active: !element.active })
+  this.setState({ data: newData })
+}
+```
+
+Additionally, because of this rendering behaviour the `render` functions in
+`columns` should only render values provided to the render function, as
```

Comment:
How about `only use values` so that we reduce the repetition of the `render` word a bit?

Reply:
Maybe we could say something like
> When a `render()` functions are provided, they must act like pure functions with respect to their props. They should only rely on their arguments to generate their output and not use any context variables). If you are using a context variable in your `render()` function, then changes on that variable will not cause a re-render of the component and will not be reflected on the table.

As per https://reactjs.org/docs/components-and-props.html#functional-and-class-components